react/modules/tasks/routing.md

```diff
@@ -3,37 +3,45 @@
 ## Application Requirements
 
 1. Add pagination:
-   - implement pagination for your existing item list
+   - implement pagination for your existing item list (search results).
    - display the current page in the browser URL using query parameters (e.g. ?page=2, e.g /search/2).
    - the pagination component should appear after receiving the list of all items.
 2. Main page displays search results. On item click page should be split into 2 section:
-   - left section will continue to display search results;
-   - right section should display details using Router Outlet (show loading indicator while making an additional call for details, add control for closing the section, also section should be closed when user clicks on the left section)
+   - left section will continue to display search results.
+   - right section should display details using Router Outlet (show loading indicator while making an additional call for details, add control for closing the section, also section should be closed when user clicks on the left section).
    - reflect in the url that "Details" section has been opened for the selected item (e.g. /?frontpage=2&details=1).
 
 ## Technical Requirements
 
-1. Create a separate branch for this task from the previous task's branch. Branch name: "hooks-and-routing".
-2. All components must be changed to **functional components**, except **Error Boundary** components, as error boundaries in React still need to be class components.
+1. Create a separate branch for this task from the previous task's branch (class-components). Branch name: "hooks-and-routing".
+2. All components must be changed to **functional components**, except the **Error Boundary** component, as error boundaries in React still need to be the class component.
    - Implement custom hook to restore search query from the local storage (LS). Use respective React lifecycle hook as a basis.
 3. All logic should be split into components:
    - If you need an access either to the component's lifecycle or the state **use hooks**.
    - All data should be stored in the **component's state**.
 4. Add routing to your application using **React Router**.
 5. Add a 404 page when user navigates to non-existing route.
-6. Add and configure a test runner: Jest or Vitest. Test runner should show the test coverage. You should aim to reach at least 80% of the test coverage.
+6. Add and configure a test runner: Jest or Vitest. Test runner should show the test coverage.
+   - You should aim to reach at least 70% of the test coverage without testing App.tsx (coverage App.tsx must be 0%).
```

Comment:
`coverage App.tsx must be 0%` -> `coverage for App.tsx must be 0%`?

Reply:
agree